features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/OtelTracerBuilderProviderTest.kt

```diff
@@ -0,0 +1,363 @@
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    @StringForgery
+    lateinit var fakeInstrumentationName: String
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        fakeServiceName = forge.anAlphabeticalString()
+        whenever(
+            mockSdkCore.getFeature(Feature.TRACING_FEATURE_NAME)
+        ) doReturn mockTracingFeatureScope
+        whenever(mockTracingFeatureScope.unwrap<TracingFeature>()) doReturn mockTracingFeature
+        whenever(mockSdkCore.getFeature(Feature.RUM_FEATURE_NAME)) doReturn mock()
+        whenever(mockSdkCore.service) doReturn fakeServiceName
+        whenever(mockSdkCore.internalLogger) doReturn mockInternalLogger
+        testedOtelTracerProviderBuilder = OtelTracerProvider.Builder(mockSdkCore)
+    }
+
+    // region feature checks
+
+    @Test
+    fun `M log a developer error W buildTracerProvider { TracingFeature not enabled }`() {
+        // GIVEN
+        whenever(mockSdkCore.getFeature(Feature.TRACING_FEATURE_NAME)) doReturn null
+
+        // WHEN
+        testedOtelTracerProviderBuilder.build()
+
+        // THEN
+        mockInternalLogger.verifyLog(
+            InternalLogger.Level.ERROR,
+            InternalLogger.Target.USER,
+            OtelTracerProvider.TRACING_NOT_ENABLED_ERROR_MESSAGE
+        )
+    }
+
+    @Test
+    fun `M log a developer error W build { default service name not available }`() {
+        // GIVEN
+        whenever(mockSdkCore.service) doReturn ""
+
+        // WHEN
+        testedOtelTracerProviderBuilder.build()
+
+        // THEN
+        mockInternalLogger.verifyLog(
+            InternalLogger.Level.ERROR,
+            InternalLogger.Target.USER,
+            OtelTracerProvider.DEFAULT_SERVICE_NAME_IS_MISSING_ERROR_MESSAGE
+        )
+    }
+
+    // endregion
+
+    // region ID generation
+
+    @Test
+    fun `M build tracers which generate Spans with 64 bits long ids W build`(
+        @StringForgery(type = StringForgeryType.ALPHA_NUMERICAL) spanName: String
+    ) {
+        // Given
+        val tracerProvider = testedOtelTracerProviderBuilder
+            .build()
+
+        // When
+        val span = tracerProvider.tracerBuilder(fakeInstrumentationName).build().spanBuilder(spanName).startSpan()
+
+        // Then
+        assertThat(span.spanContext.spanIdBytes.size).isEqualTo(8)
+    }
+
+    @Test
+    fun `M build tracers which generate Spans with 128 bits long trace ids W build`(
+        @StringForgery(type = StringForgeryType.ALPHA_NUMERICAL) spanName: String
+    ) {
+        // Given
+        val tracerProvider = testedOtelTracerProviderBuilder.build()
+
+        // When
+        val span = tracerProvider.tracerBuilder(fakeInstrumentationName).build().spanBuilder(spanName).startSpan()
+        val traceId = span.spanContext.traceId
+        // Then
+        assertThat(traceId.substring(0, 16)).isEqualTo("0000000000000000")
+        assertThat(span.spanContext.traceIdBytes.size).isEqualTo(16)
```

Comment:
so with Otel by default we have 64 bits for span ID, but 128 bits for Trace ID?

Reply:
yes that's the way it is and the first 64bits are 00 for the trace id. Our intake for RUM spans does not support that yet (it should but based on my tests it doesn't yet) so I will have to remove later in the mapper these 64 bits from the trace id before serializing it.